langwatch/src/pages/settings/teams/[team].tsx

```diff
@@ -24,29 +91,92 @@ export default function EditTeamPage() {
       slug: teamSlug as string,
       organizationId: organization?.id ?? "",
     },
-    { enabled: typeof teamSlug === "string" && !!organization?.id }
+    { enabled: typeof teamSlug === "string" && !!organization?.id },
   );
 
-  if (!team.data) return <SettingsLayout />;
+  // Handle UNAUTHORIZED error first
+  if (team.error) {
+    const error = team.error;
+    if (
+      error instanceof TRPCClientError &&
+      error.data?.code === "UNAUTHORIZED"
+    ) {
+      return (
+        <SettingsLayout>
+          <VStack paddingX={4} paddingY={6} gap={4} align="start">
+            <PermissionAlert
+              permission="team:view"
+              message="You don't have permission to view this team. Please contact your team administrator for access."
+            />
+          </VStack>
+        </SettingsLayout>
+      );
+    }
+  }
+
+  // Handle loading state
+  if (team.isLoading || !team.data) {
+    return (
+      <SettingsLayout>
```

Comment:
I feel like, if you have large chunks of markup behind if statements, it's a good sign that we can break these out into smaller UI components.